packages/labs/task/src/task.ts

```diff
@@ -226,17 +246,19 @@ export class Task<
             // Ignore user errors from onComplete.
           }
           this.status = TaskStatus.COMPLETE;
-          this._resolveTaskComplete(result as R);
+          this._resolveTaskComplete?.(result as R);
         } else {
           try {
             this._onError?.(error);
           } catch {
             // Ignore user errors from onError.
           }
           this.status = TaskStatus.ERROR;
-          this._rejectTaskComplete(error);
+          this._rejectTaskComplete?.(error);
         }
         this._value = result as R;
+        // The run's ended so don't generate a task complete promise and either
+        // return the existing, resolved promise or a new resolved promise.
```

Comment:
Is this comment still valid?

I think this is the spot where you can do:

```ts
this._resolveTaskComplete = undefined;
this._rejectTaskComplete = undefined;
this._taskComplete = undefined;
```

Reply:
oops not valid anymore. No longer necessary since I do that inthe taskComplete getter.